client retrieve fail

#lotus client retrieve bafkreicjzjoycbkp3uqfoiuuxe2qwyc5axqn7eo2bgsg7of5476wyhaxfu t.txt
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/filecoin-project/lotus/cli.glob..func12(0xc0002e8e10, 0x0, 0x0)
	/home/wanghui/go/src/github.com/filecoin-project/lotus/cli/client.go:233 +0x6eb
gopkg.in/urfave/cli%2ev2.(*Command).Run(0x2bf7020, 0xc0002e8db0, 0x0, 0x0)
	/home/wanghui/go/pkg/mod/gopkg.in/urfave/cli.v2@v2.0.0-20180128182452-d3ae77c26ac8/command.go:167 +0x6cf
gopkg.in/urfave/cli%2ev2.(*App).RunAsSubcommand(0xc000570480, 0xc0002e8ba0, 0x0, 0x0)
	/home/wanghui/go/pkg/mod/gopkg.in/urfave/cli.v2@v2.0.0-20180128182452-d3ae77c26ac8/app.go:381 +0x907
gopkg.in/urfave/cli%2ev2.(*Command).startApp(0x2bf7380, 0xc0002e8ba0, 0x2c2c796, 0x1)
	/home/wanghui/go/pkg/mod/gopkg.in/urfave/cli.v2@v2.0.0-20180128182452-d3ae77c26ac8/command.go:248 +0x646
gopkg.in/urfave/cli%2ev2.(*Command).Run(0x2bf7380, 0xc0002e8ba0, 0x0, 0x0)
	/home/wanghui/go/pkg/mod/gopkg.in/urfave/cli.v2@v2.0.0-20180128182452-d3ae77c26ac8/command.go:85 +0xd18
gopkg.in/urfave/cli%2ev2.(*App).Run(0xc000570300, 0xc000032050, 0x5, 0x5, 0x0, 0x0)
	/home/wanghui/go/pkg/mod/gopkg.in/urfave/cli.v2@v2.0.0-20180128182452-d3ae77c26ac8/app.go:261 +0x68c
main.main()
	/home/wanghui/go/src/github.com/filecoin-project/lotus/cmd/lotus/main.go:67 +0x67b

--- a/cli/client.go
+++ b/cli/client.go
@@ -230,6 +230,10 @@ var clientRetrieveCmd = &cli.Command{
 
 		// TODO: parse offer strings from `client find`, make this smarter
 
+		if len(offers) < 1 {
+			fmt.Println("Failed to find file")
+			return nil
+		}
 		order := offers[0].Order()
 		order.Client = payer
 

